File: 1524-string-matching-in-an-array/string-matching-in-an-array.cpp
class Solution {
public:
    vector<string> stringMatching(vector<string>& words) {
        set<string> res;
        for(int i = 0; i < words.size(); i++){
            for(int j = i + 1; j < words.size(); j++){
                if(words[i] == words[j]){
                    continue;
                }
                else if(words[i].find(words[j]) != string::npos){
                    res.insert(words[j]);
                }
                else if(words[j].find(words[i]) != string::npos){
                    res.insert(words[i]);
                }
            }
        }
        return vector<string>(res.begin(), res.end());
    }
};

File: 1524-string-matching-in-an-array/string-matching-in-an-array.py
class Solution:
    def stringMatching(self, words: List[str]) -> List[str]:
        k=[]
        for i in range(0,len(words)):
            for j in range(i+1,len(words)):
                if(words[i] == words[j]):
                    pass
                elif(words[i] in words[j]):
                    k.append(words[i])
                elif(words[j] in words[i]):
                    k.append(words[j])
                else:
                    pass
        return list(set(k))

